internal/compiler/passes/const_propagation.rs

```diff
@@ -105,6 +105,19 @@ fn simplify_expression(expr: &mut Expression) -> bool {
             }
             can_inline
         }
+        Expression::UnaryOp { sub, op } => {
```

Comment:
Nice improvements in this pass!

btw, there is a test at the bottom of this file, and it would be nice if it could be updated to include some more complex things that you now handle

Reply:
Extended the test with the new functionality.